Tools/clinic/libclinic/parse_args.py

```diff
@@ -520,28 +527,15 @@ def parse_pos_only(self) -> None:
             if p.is_vararg():
                 if self.fastcall:
                     parser_code.append(libclinic.normalize_snippet("""
-                        %s = PyTuple_New(%s);
-                        if (!%s) {{
-                            goto exit;
-                        }}
-                        for (Py_ssize_t i = 0; i < %s; ++i) {{
-                            PyTuple_SET_ITEM(%s, i, Py_NewRef(args[%d + i]));
-                        }}
+                        %s = args + %s;
                         """ % (
                             p.converter.parser_name,
-                            left_args,
-                            p.converter.parser_name,
-                            left_args,
-                            p.converter.parser_name,
-                            self.max_pos
+                            self.vararg
                         ), indent=4))
                 else:
                     parser_code.append(libclinic.normalize_snippet("""
-                        %s = PyTuple_GetSlice(%d, -1);
-                        """ % (
-                            p.converter.parser_name,
-                            self.max_pos
-                        ), indent=4))
+                        %s = _PyTuple_CAST(args)->ob_item;
+                        """ % p.converter.parser_name, indent=4))
```

Comment:
I'd like if we could refactor this a little bit more nice, now that we've got the chance. Something like this, perhaps:

```diff
diff --git a/Tools/clinic/libclinic/parse_args.py b/Tools/clinic/libclinic/parse_args.py
index db26f084e42..2b73f78822a 100644
--- a/Tools/clinic/libclinic/parse_args.py
+++ b/Tools/clinic/libclinic/parse_args.py
@@ -525,17 +525,13 @@ def parse_pos_only(self) -> None:
         use_parser_code = True
         for i, p in enumerate(self.parameters):
             if p.is_vararg():
+                var = p.converter.parser_name
                 if self.fastcall:
-                    parser_code.append(libclinic.normalize_snippet("""
-                        %s = args + %s;
-                        """ % (
-                            p.converter.parser_name,
-                            self.vararg
-                        ), indent=4))
+                    code = f"{var} = args + {self.vararg};"
                 else:
-                    parser_code.append(libclinic.normalize_snippet("""
-                        %s = _PyTuple_CAST(args)->ob_item;
-                        """ % p.converter.parser_name, indent=4))
+                    code = f"{var} = _PyTuple_CAST(args)->ob_item;"
+                formatted_code = libclinic.normalize_snippet(code, indent=4)
+                parser_code.append(formatted_code)
                 continue
 
             displayname = p.get_displayname(i+1)

```